### ISSUE TITLE
ITMS-90338: Non-public API usage

### ISSUE THREAD

**Issue by nitrique** [react-native-webrtc/react-native-webrtc#1129]
Hi,

Updated to release 1.98.0 yesterday and now on build we have the following from Apple:

> ITMS-90338: Non-public API usage - The app contains one or more corrupted binaries. Rebuild the app and resubmit.,The app references non-public selectors in ensweet: addStream:, initWithBuffer:rotation:timeStampNs:, initWithURLStrings:, isBinary, isPassthrough, localDescription, newSocketQueueForConnectionFromAddress:onSocket:, onSuccess:, removeValuesForKeys:completion:, sdp, sendData:, setChannelId:, socket:didAcceptNewSocket:, socket:didConnectToHost:port:, socket:didConnectToUrl:, socket:didReadData:withTag:, socket:didReadPartialDataOfLength:tag:, socket:didReceiveTrust:completionHandler:, socket:didWriteDataWithTag:, socket:didWritePartialDataOfLength:tag:, socket:shouldTimeoutReadWithTag:elapsed:bytesDone:, socket:shouldTimeoutWriteWithTag:elapsed:bytesDone:, socketDidCloseReadStream:, socketDidDisconnect:withError:, socketDidSecure:, stringForType:, typeForString:, videoSource. If method names in your source code match the private Apple APIs listed above, altering your method names will help prevent this app from being flagged in future submissions. In addition, note that one or more of the above APIs may be located in a static library that was included with your app. If so, they must be removed. For further information, visit the Technical Support Information at http://developer.apple.com/support/technical/

Best,
Nicolas

**Comment by nitrique** [comment]
No problem with downgrade on 1.94, problem with 1.98... 
So not Apple matter as you mention in other issue.

Please don't be so expeditive...

**Comment by goguda** [comment]
> No problem with downgrade on 1.94, problem with 1.98... So not Apple matter as you mention in other issue.
> 
> Please don't be so expeditive...


> We've been through this before, that's why I know the framework is not at fault.

I'm also getting this after upgrading to 1.98 from 1.94. I don't think it's fair at all to completely close and dismiss the issue considering it **is** this framework that is causing the issue.

Contacting Apple support does nothing, they want to have nothing to do with fixing these kind of issues when they are introduced from an external library such as this one. They aren't going to give the OK and go ahead on this just because we contact them and say they're wrong.

Saying so is just dismissing the fact there is a problem and will get us nowhere, especially as it starts to affect more people who upgrade this library.

Downgrading to 1.94 fixes it for now, but I kindly ask that the issue be reopened so it can be properly investigated.

**Comment by saghul** [comment]
We got hit by the same error on the latest submission, but my answer remains the same: it's a false positive.

I've raised a TSI case and will update with the progress here.

For future reference, here is a breakdown of those "private" APIs:

> Frameworks/JitsiMeetSDK.framework/JitsiMeetSDK: addStream:, initWithBuffer:rotation:timeStampNs:, initWithURLStrings:, isBinary, isPassthrough, localDescription, onSuccess:, removeValuesForKeys:completion:, sdp, sendData:, setCategoryOptions:, setChannelId:, setConfiguration:error:, showLoading, streamId, stringForType:, typeForString:, videoSource. 

All of them come from WebRTC, which I suspect triggers the false positive because Safari also has WebRTC APIs, but they are not exposed as frameworks.

**Comment by PrimulaX** [comment]
That happens when I build using Hermes on in Podfile in my case.

**Comment by saghul** [comment]
I am not using Hermes, FWIW.

**Comment by PrimulaX** [comment]
@saghul It's weird, I have ` "react-native-webrtc": "1.98.0"` and Apple doesn’t have a problem without Hermes.
My reject notice is exactly same
<img width="683" alt="Screenshot 2022-03-21 at 12 35 50" src="https://user-images.githubusercontent.com/16132259/159227946-c27390ba-2e5a-4219-962f-3b96867f4737.png">

**Comment by saghul** [comment]
Another interesting this is you seem to have more APIs listed there... As an update, I'm still waiting on my TSI response...

**Comment by andybui090** [comment]
Any update, apple can not approve, so I using version 0.94 to resolve

**Comment by saghul** [comment]
Not yet, I had to open another ticket :-/

**Comment by saghul** [comment]
Still investigating.

Those of you getting rejected: are you using bitcode?

**Comment by PrimulaX** [comment]
@saghul Yes.

**Comment by saghul** [comment]
Success! I manages to get a TestFlight build pass approval. Unfortunately it required many hacks, but at least they are on the build side so they don't affect apps.

I'll wrap up the M100 release with these fixes.

**Comment by saghul** [comment]
Alright, 1.100.0 is out: https://github.com/react-native-webrtc/react-native-webrtc/releases/tag/1.100.0 please let me know how it goes.

**Comment by moisestecnosul** [comment]
It works perfectly, @saghul ! I'm very glad. Thanks! How can we gratify you? You really helped. Again, thanks!

**Comment by saghul** [comment]
> It works perfectly, @saghul ! I'm very glad. Thanks! How can we gratify you? You really helped. Again, thanks!

Thanks for confirming! ❤️

**Comment by vinhluong** [comment]
Shut your fuckup

**Comment by saghul** [comment]
If you saw the now hidden comment, the user is now blocked from this organization.
